internal/dataapi/listener.go

```diff
@@ -85,6 +86,13 @@ func (lis *Listener) Run(ctx context.Context) {
 		}
 	}()
 
+	// Shouldn't we also call lis.lis.Close()?
```

Comment:
`srv.Serve(lis.lis)` closes listener upon returning. I don't think it's necessary.